server/hedley/modules/custom/hedley_stats/hedley_stats.module

```diff
@@ -313,69 +319,136 @@ function hedley_stats_calculate_stats_worker($queue_item) {
 }
 
 /**
- * Get all the stats for each health center.
+ * Generate and store in cache health center statistics data.
  *
- * We don't worry about caching here because each function has its own caching.
+ * @param int $health_center_id
+ *   The health center node ID.
+ *
+ * @throws EntityMetadataWrapperException
+ */
+function hedley_stats_calculate_stats_for_health_center($health_center_id) {
+  // We started recalculation of HC statistics. Therefore, we delete its
+  // statistics hash cache, to indicate that it's not available (until
+  // recalculation is completed).
+  hedley_stats_handle_cache(HEDLEY_STATS_CACHE_CLEAR, HEDLEY_STATS_SYNC_STATS_CACHE, $health_center_id);
+
+  // Get all clinics of type 'FBF' for the HC.
+  $fbf_clinics = hedley_health_center_get_clinics_by_health_center($health_center_id, 'fbf');
+  // Calculate statistics that are provided for FBF clinics only.
+  if (!empty($fbf_clinics)) {
+    hedley_stats_get_session_attendance_stats_by_health_center($health_center_id, $fbf_clinics);
+  }
+
+  $villages_with_residents = hedley_stats_get_villages_with_residents($health_center_id);
+
+  hedley_stats_get_case_management($health_center_id);
+  hedley_stats_get_children_beneficiaries_stats_by_period($health_center_id);
+  hedley_stats_get_family_planning_stats_by_period($health_center_id, HEDLEY_STATS_PERIOD_PAST_THREE_MONTHS);
+  hedley_stats_get_total_encounters($health_center_id, $villages_with_residents);
+  hedley_stats_get_acute_illness_data($health_center_id);
+  hedley_stats_get_prenatal_data($health_center_id);
+
+  $stats_cache_hash = md5(time());
+  // Store in cache only the hash. All statistics have their own cache.
+  hedley_stats_handle_cache(HEDLEY_STATS_CACHE_SET, HEDLEY_STATS_SYNC_STATS_CACHE, $health_center_id, NULL, $stats_cache_hash);
+
+  // Free up memory.
+  drupal_static_reset();
+}
+
+/**
+ * Pull from cache and format statistics data for a health center.
  *
  * @param int $health_center_id
  *   The health center node ID.
- * @param string $cached_hash
- *   Optional; If non empty value provided, we do not update cache hash.
  *
  * @return array
- *   The stats or error details if anything went wrong.
+ *   Statistics data for a health center.
+ *
+ * @throws EntityMetadataWrapperException
  */
-function hedley_stats_calculate_stats_for_health_center($health_center_id, $cached_hash = '') {
-  try {
-    $wrapper = entity_metadata_wrapper('node', $health_center_id);
-
-    // Get all clinics of type 'FBF' for the HC.
-    $fbf_clinics = hedley_health_center_get_clinics_by_health_center($health_center_id, 'fbf');
-    // Calculate statistics that are provided for FBF clinics only.
-    $completed_program = $missed_sessions = [];
-    if (!empty($fbf_clinics)) {
-      list($completed_program, $missed_sessions) = hedley_stats_get_session_attendance_stats_by_health_center($health_center_id, $fbf_clinics);
-    }
+function hedley_stats_pull_stats_for_health_center($health_center_id) {
```

Comment:
@anvmn I'm not sure if the logic defined here is correct. My understanding is if there is no stats for any of the following cache_data the function will return an empty array. Could be possible to have a null value for any of the following `hedley_stats_handle_cache` and still want to treat it as a valid value?

If this is not the case. The PR looks good.

Reply:
@mariano-dagostino 
Any of those returnung NULL means that HC statistics are being recalculated by server.
In this case, we return empty array, indicating that statistics are not available at current moment.